src/WorkflowGui/Controller/WorkflowController.php

```diff
@@ -309,4 +313,60 @@ protected function sanitizeConfiguration($configuration)
 
         return $configuration;
     }
+
+    public function visualizeAction(Request $request)
+    {
+        $this->isGrantedOr403();
+
+        try {
+            $dotExecutable = Console::getExecutable('dot');
+        } catch(\Exception $e) {
+            return new Response('Please install graphviz to visualize workflows');
+        }
+
+        try {
+            $cliCommand = '"'.Console::getPhpCli().'" "'.PIMCORE_PROJECT_ROOT . '/bin/console" pimcore:workflow:dump '.$request->get('workflow').' | "'.$dotExecutable.'" -Tsvg';
```

Comment:
duplicate code, please avoid that. Use a separate function to create the image

Reply:
Refactored this.